auth/auth/auth.py

```diff
@@ -53,13 +76,153 @@ async def get_index(request):  # pylint: disable=unused-argument
     return aiohttp.web.HTTPFound(deploy_config.external_url('auth', '/login'))
 
 
+@routes.get('/signup')
+@web_maybe_authenticated_user
+async def signup(request, userdata):
+    db = request.app['db']
+
+    next = request.query.get('next', deploy_config.external_url('notebook', ''))
+
+    session = await aiohttp_session.get_session(request)
+
+    if 'pending' in session:
+        email = session['email']
+        user = await user_from_email(db, email)
+
+        if user is None:
+            cleanup_session(session)
+            return aiohttp.web.HTTPFound(next)
+
+        if user['state'] == 'deleting' or user['state'] == 'deleted':
+            cleanup_session(session)
+            set_message(session, f'Account for {user["username"]} is deleted.', 'error')
+            return aiohttp.web.HTTPFound(next)
```

Comment:
I feel like this shouldn't go to `next`, the user has no account, they can't possibly do anything. Shouldn't we bounce back to notebook.hail.is?

Reply:
Or maybe an error page specifically about your account being deleted?